### PR TITLE
ENG-1453 - wasm: Keyword PII matching

### PR DESCRIPTION
**Pull request by blinktag**

### CODE REVIEW

File: libs/wasm-detective/src/keywords/scanner.rs [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+                value: f.value.clone(),
+                confidence: def.score as f32,
+                entity: def.entity.clone(),
+            };
+
+            matches.push(m);
+            return matches;
+        }
+
+        // Partial match check here
```
Comment (dselans) [inline libs/wasm-detective/src/keywords/scanner.rs:217]: Hmm.. maybe this should be configurable in the step config - ie. something like `EnablePartialPathMatching` - that way we don't take the hit by the default.

File: libs/wasm-detective/src/keywords/test_scanner.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    let mut pii = FieldPII::new(random_strings);
+
+    b.iter(|| {
+        let _ = pii.scan(payload);
+    });
+}
+
+#[bench]
+fn bench_standard_pii(b: &mut Bencher) {
```
Comment (dselans) [inline libs/wasm-detective/src/keywords/test_scanner.rs:88]: This is really great - let's include the benchmark results in some file. Maybe add a `make` target for `bench` that writes the file? This way we can always look it up / link it to folks.

Comment (dselans) [reply libs/wasm-detective/src/keywords/test_scanner.rs:88]: Also, let's do a few of these - with different size JSON payloads - one with 1KB JSON, 10KB, 100KB.

**Review by dselans — APPROVED** [review]
Added a couple of minor comments/questions. Nicely done Mark! Can you demo this tomorrow morning?

File: libs/wasm-detective/src/keywords/test_scanner.rs (hunk 2 above)
Comment (blinktag) [reply libs/wasm-detective/src/keywords/test_scanner.rs]: Good idea, done!

File: libs/wasm-detective/src/keywords/scanner.rs (hunk 1 above)
Comment (blinktag) [reply libs/wasm-detective/src/keywords/scanner.rs:217]: Let's architect the protos for this. We'll also need fields in the future for custom keyword lists, so best to do it properly now

**Pull request merged**